transaction/src/action/swap.rs

```diff
@@ -0,0 +1,105 @@
+use anyhow::Error;
+
+use penumbra_crypto::rdsa::{Signature, SpendAuth};
+use penumbra_crypto::NotePayload;
+use penumbra_crypto::{proofs::transparent::SpendProof, MockFlowCiphertext};
+use penumbra_crypto::{value, SwapCiphertext};
+use penumbra_proto::dex::TradingPair;
+use penumbra_proto::{dex as pb, Protobuf};
+
+#[derive(Clone, Debug)]
+pub struct Swap {
+    // A proof that this is a valid state change.
+    pub proof: SpendProof,
```

Comment:
We shouldn't use a `SpendProof` here, because the swap proof statement is different.  Instead we should make a new `SwapProof` (it can be a placeholder at the moment), and fill in the swap-specific proof statements there later.